src/macaron/slsa_analyzer/package_registry/pypi_registry.py

```diff
@@ -305,3 +271,146 @@ def get_maintainer_join_date(self, username: str) -> datetime | None:
         res: datetime | None = parse_datetime(datetime_val, datetime_format)
 
         return res.replace(tzinfo=None) if res else None
+
+
+@dataclass
+class PyPIPackageJsonAsset:
```

Comment:
I wonder if there is any reason why `PyPIPackageJsonAsset` doesn't inherit from `NamedTuple` compared to https://github.com/oracle/macaron/blob/da182aaa537367c85625f37160da570d9952c975/src/macaron/slsa_analyzer/package_registry/jfrog_maven_registry.py#L26

Reply:
`PyPIPackageJsonAsset` is not immutable because the `package_json` attribute will be set upon download. So, `NamedTuple` cannot be used.